src/main/java/org/opensearch/security/resources/ResourceSharingIndexHandler.java

```diff
@@ -152,16 +157,17 @@ public ResourceSharing indexResourceSharing(String resourceId, String resourceIn
         try (ThreadContext.StoredContext ctx = this.threadPool.getThreadContext().stashContext()) {
             ResourceSharing entry = new ResourceSharing(resourceIndex, resourceId, createdBy, shareWith);
 
-            IndexRequest ir = client.prepareIndex(resourceSharingIndex)
+            IndexRequest ir = client.prepareIndex(getSharingIndex(resourceIndex))
```

Comment:
we should extract `getSharingIndex(resourceIndex)` and declare it as method level variable.